frontend/pages/model/[modelId]/release/new.tsx

```diff
@@ -55,15 +61,47 @@ export default function NewRelease() {
       }
 
       const metadata = filesMetadata.find((fileWithMetadata) => fileWithMetadata.fileName === file.name)?.metadata
-      const postFileResponse = await postFile(file, model.id, file.name, file.type, metadata)
 
-      if (!postFileResponse.ok) {
-        setErrorMessage(await getErrorMessage(postFileResponse))
+      const fileResponse = await axios
+        .post(
+          metadata
+            ? `/api/v2/model/${model.id}/files/upload/simple?name=${file.name}&mime=${file.type}?${qs.stringify({
+                metadata,
+              })}`
+            : `/api/v2/model/${model.id}/files/upload/simple?name=${file.name}&mime=${file.type}`,
+          file,
+          {
+            onUploadProgress: function (progressEvent) {
+              if (progressEvent.total) {
+                const percentCompleted = Math.round((progressEvent.loaded * 100) / progressEvent.total)
+                setCurrentFileUploadProgress({ fileName: file.name, uploadProgress: percentCompleted })
+              }
+            },
+          },
+        )
+        .catch(function (error) {
+          if (error.response) {
+            sendNotification({
```

Comment:
If we split this out into `actions/release` as described in another comment, we can just reuse the same logic as was here before for handling errors. The action function can be responsible for returning an error in a format we expect and this component doesn't have to do anything special.